qiskit/providers/honeywell/credentials/__init__.py

```diff
@@ -42,69 +43,87 @@ class HoneywellCredentialsError(HoneywellError):
+        if proxies is not None:
+            self.proxies = proxies
+
+    def load_from_environ(self):
+        """ Attempts to read credentials from environment variable """
+        self.token = os.getenv('HON_QIS_API')
+
+    def load_from_qiskitrc(self, filename):
+        """ Attempts to read credentials from qiskitrc file
+        The default qiskitrc location is in ``$HOME/.qiskitrc/qhprc``
+        """
+        config_parser = ConfigParser()
         try:
-            creds = reader_func(**kwargs)
-            if creds:
-                break
-        except HoneywellCredentialsError as ex:
-            logger.warning('Automatic discovery failed: %s', str(ex))
-    return creds
-
-
-def read_creds_from_environ():
-    """ Attempts to read credentials from environment variable """
-    return os.getenv('HON_QIS_API')
-
-
-def read_creds_from_qiskitrc(filename):
-    """ Attempts to read credentials from qiskitrc file
-    The default qiskitrc location is in ``$HOME/.qiskitrc/qhprc``
-    """
-    config_parser = ConfigParser()
-    try:
-        config_parser.read(filename)
-    except ParsingError as ex:
-        raise HoneywellCredentialsError(str(ex))
-    return config_parser.get(SECTION_NAME, 'API_KEY', fallback=None)
-
-
-def write_creds_to_qiskitrc(token, overwrite=False, filename=DEFAULT_QISKITRC_FILE):
-    """ Stores the credentials to qiskitrcc file
-    The default qiskitrc location is in ``$HOME/.qiskitrc/qhprc``
-    """
-    config_parser = ConfigParser()
-    try:
-        config_parser.read(filename)
-    except ParsingError as ex:
-        raise HoneywellCredentialsError(str(ex))
-    if config_parser.has_section(SECTION_NAME) and \
-       config_parser.get(SECTION_NAME, 'API_KEY') and not overwrite:
-        return
-    config_parser[SECTION_NAME] = {'API_KEY': token}
-    with open(filename, 'w') as conf_file:
-        config_parser.write(conf_file)
-
-
-def remove_creds_from_qiskitrc(filename=DEFAULT_QISKITRC_FILE):
-    """ Removes the credentials from the configuration file
-    The default qiskitrc location is in ``$HOME/.qiskitrc/qhprc``
-    """
-    config_parser = ConfigParser()
-    try:
-        config_parser.read(filename)
-    except ParsingError as ex:
-        raise HoneywellCredentialsError(str(ex))
-    if not (config_parser.has_section(SECTION_NAME) and config_parser.get(SECTION_NAME,
-                                                                          'API_KEY')):
-        return
-    config_parser.remove_option(SECTION_NAME, 'API_KEY')
-    with open(filename, 'w') as conf_file:
-        config_parser.write(conf_file)
+            config_parser.read(filename)
+        except ParsingError as ex:
+            raise HoneywellCredentialsError(str(ex))
+
+        setattr(self, 'token',
+                config_parser.get(SECTION_NAME, 'API_KEY', fallback=self.token))
+        setattr(self, 'proxies',
+                literal_eval(config_parser.get(SECTION_NAME, 'proxies', fallback='{}')))
```

Comment:
This seems like an odd pattern, how do you expect users to specify a proxy? Maybe having something in the README about how proxies are supposed to be set.

I assume it's just to do the list parsing so you can specify multiple proxies and get a tuple or list from the string. But using the ast parser for this seems both heavyweight and a potential sec risk (because the ast parser will do more than just split a list). I think it would be better to just say the list of proxies have to be a comma separated string and just do 
```
proxies_str = config_parser.get(SECTION_NAME, 'proxies', fallback='{}')
setattr(self, 'proxies', [x.strip() for x in proxies_str.split(',')])
```

But either way we should document what we want the proxy config format to be in the readme.

Reply:
https://github.com/Qiskit/qiskit-ibmq-provider/blob/master/qiskit/providers/ibmq/credentials/configrc.py#L75

I was trying to be similar to how the IBM backend was handling this feature, but I also agree it seems like a big hammer for a small problem.  I will add documentation to show proper configuration, and consider alternatives.

Reply:
Switch to json dictionaries to avoid any issues.  Examples on configuration provided in documentation.